examples/StockDonor/src/StockDonorColumn.tsx

```diff
@@ -1,33 +1,60 @@
-import React from 'react';
+import React, { useEffect } from 'react';
 import {
+  ArrayElement,
   BasicCellLayout,
   CellProps,
   ColumnDefinition,
+  create,
+  PluginDataStore,
+  Plugins,
   QueryClientProviderProxy,
-  RecordWithId,
 } from '@openmsupply-client/common';
 import { StockLineRowFragment } from '@openmsupply-client/system';
 import { usePluginData } from './api';
 
-const DonorColumn = <T extends RecordWithId>({ rowData }: CellProps<T>) => {
-  const { data } = usePluginData.data(rowData?.id ?? '');
+const useColumnStore = create<PluginDataStore<StockLineRowFragment, string>>(
+  (set, get) => ({
+    data: [],
+    set: data => set(state => ({ ...state, data })),
+    getById: row =>
+      get().data.find(({ relatedRecordId }) => relatedRecordId == row.id),
+  })
+);
+
+type StockDonorColumn = NonNullable<ArrayElement<Plugins['stockColumn']>>;
+
+export const StateLoader: ArrayElement<StockDonorColumn['StateLoader']> = ({
```

Comment:
This is an example of using StateLoader, to load all data for new column for visible stock lines, rather then calling a query on each stock line (which is what we were doing previously)

Reply:
There is explanation of this in the docs pr